src/main/java/org/ice4j/util/PacketQueue.java

```diff
@@ -279,32 +280,39 @@ private void doAdd(T pkt)
         if (closed)
             return;
 
-        synchronized (queue)
+        while (!queue.offer(pkt))
         {
-            while (!queue.offer(pkt))
+            // Drop from the head of the queue.
+            T p = queue.poll();
+            if (p != null)
             {
-                // Drop from the head of the queue.
-                T p = queue.poll();
-                if (p != null)
+                if (queueStatistics != null)
                 {
-                    if (queueStatistics != null)
-                    {
-                        queueStatistics.remove(System.currentTimeMillis());
-                    }
-                    if (logDroppedPacket(++numDroppedPackets))
-                    {
-                        logger.warning(
-                            "Packets dropped (id=" + id + "): " + numDroppedPackets);
-                    }
+                    queueStatistics.remove(System.currentTimeMillis());
+                }
+                final int numDroppedPackets =
+                    this.numDroppedPackets.incrementAndGet();
```

Comment:
so it looks like this doesn't get accessed from anywhere else, is that right?  does this method get called from multiple threads?

Reply:
@bbaldino it might be, there is no protection against calling `PacketQueue.add` from multiple threads.